manifests/database/postgresql.pp

```diff
@@ -24,6 +28,62 @@
       port                    => scanf($database_port, '%i')[0],
     }
 
+    # configure PostgreSQL communication with Puppet Agent SSL certificates if
+    # postgresql_ssl_on is set to true
+    if $postgresql_ssl_on {
+      file {'postgres private key':
+        ensure  => present,
+        path    => "${postgresql::params::datadir}/server.key",
+        source  => $postgresql_ssl_key_path,
+        owner   => 'postgres',
+        mode    => '0600',
+        require => Package['postgresql-server'],
+      }
+
+      concat {'postgres cert bundle':
+        ensure  => present,
+        path    => "${postgresql::params::datadir}/server.crt",
+        owner   => 'postgres',
+        require => Package['postgresql-server'],
+      }
+
+      concat::fragment {'agent cert':
+        target => 'postgres cert bundle',
+        source => $postgresql_ssl_cert_path,
+        order  => '1',
+      }
+
+      concat::fragment {'CA bundle':
+        target => 'postgres cert bundle',
+        source => $postgresql_ssl_ca_cert_path,
+        order  => '2',
+      }
+
+      postgresql::server::config_entry {'ssl':
+        ensure  => present,
+        value   => 'on',
+        require => [File['postgres private key'], Concat['postgres cert bundle']]
+      }
+
+      postgresql::server::config_entry {'ssl_key_file':
+        ensure  => present,
+        value   => "${postgresql::params::datadir}/server.key",
+        require => [File['postgres private key'], Concat['postgres cert bundle']]
+      }
+
+      postgresql::server::config_entry {'ssl_cert_file':
+        ensure  => present,
+        value   => "${postgresql::params::datadir}/server.crt",
+        require => [File['postgres private key'], Concat['postgres cert bundle']]
+      }
+
+      # postgresql::server::config_entry {'ssl_ca_file':
```

Comment:
I eventually got to the state where postgresql required this setting with the error message
```
2021-03-08 22:23:15.457 UTC [17580] LOG:  client certificates can only be checked if a root certificate store is available
2021-03-08 22:23:15.457 UTC [17580] HINT:  Make sure the configuration parameter "ssl_ca_file" is set.
2021-03-08 22:23:15.457 UTC [17580] CONTEXT:  line 31 of configuration file "/etc/postgresql/9.6/main/pg_hba.conf"
2021-03-08 22:23:15.457 UTC [17580] LOG:  client certificates can only be checked if a root certificate store is available
2021-03-08 22:23:15.457 UTC [17580] HINT:  Make sure the configuration parameter "ssl_ca_file" is set.
2021-03-08 22:23:15.457 UTC [17580] CONTEXT:  line 36 of configuration file "/etc/postgresql/9.6/main/pg_hba.conf"
2021-03-08 22:23:15.457 UTC [17580] FATAL:  could not load pg_hba.conf
```